### PR TITLE
add sengled zigee-contact and zigbee-motion-sensor

### PR DESCRIPTION
**Pull request by sengled-desi**
zigee-contact:
add the init.lua file to the path ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-contact\src\sengled.
modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-contact\fingerprints.yml file, a new device type E2D-G73 is added.
zigbee-motion-sensor:
add the init.lua file to the path ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-motion-sensor\src\sengled.
modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-motion-sensor\fingerprints.yml file, a new device type E1M-G7H is added.

### CODE REVIEW

**Review by greens — COMMENTED** [review]
Can you respond to my comment on the original PR:

>If you use this method, the battery voltage attribute will be configured for you using the defaults. The only difference between your config and the defaults is the max reporting interval, which is 21600 in the defaults.

>Similarly, your IAS Zone configuration is not that far removed from the defaults (a lower max interval of 300) and a nil reportable_change.

>I think, if you wanted, all you'd have to have is the linear_voltage_init. You could use tests to confirm that the results of the configure lifecycle event are what you expect.

**Review by greens — CHANGES_REQUESTED** [review]
This code is never included in the base drivers and will never be loaded as written.

**Comment by lelandblue** [comment]
Hello @sengled-desi 

Please see our comments above. We need you to take action before WWST Certification can proceed. 1. Please see the comments above from @greens and make the changes recommended. 2. Please sign the CLA, this needs to be signed before we can merge the code. 

Thank you - if you have questions please reach out to partners@smartthings.com

File: drivers/SmartThings/zigbee-contact/src/init.lua [hunk 1]
```diff
@@ -49,7 +49,8 @@ local zigbee_contact_driver_template = {
     require("aurora-contact-sensor"),
     require("contact-temperature-sensor"),
     require("multi-sensor"),
-    require("smartsense-multi")
+	require("smartsense-multi"),
```
Comment (greens) [inline drivers/SmartThings/zigbee-contact/src/init.lua:52]: spacing

File: drivers/SmartThings/zigbee-motion-sensor/src/init.lua [hunk 2]
```diff
@@ -42,7 +42,8 @@ local zigbee_motion_driver = {
     require("centralite"),
     require("smartthings"),
     require("smartsense"),
-    require("thirdreality")
+	require("thirdreality"),
```
Comment (greens) [inline drivers/SmartThings/zigbee-motion-sensor/src/init.lua:45]: same

**Review by greens — CHANGES_REQUESTED** [review]
Write some tests please.

**Comment by sengled-desi** [comment]
@greens @lelandblue  Sorry, I was careless when modifying the file, this time the modification should be okay!

**Review by greens — CHANGES_REQUESTED** [review]
Would still like to see some tests. They don't need to to be big, since you're mostly just changing how the device responds to `doConfigure`, but we want to make sure that that behavior is not changed.

**Comment by lelandblue** [comment]
Hey @sengled-desi Just a reminder in order for this to proceed please provide unit test.

**Comment by sengled-desi** [comment]
@lelandblue  We have submitted several other products for certification earlier, but they all passed, this product is consistent with other products, I don't really clear what problems it has, if I need to submit some test units, what tools do I need to use to test which case

**Comment by greens** [comment]
@sengled-desi hopefully this helps: https://developer.smartthings.com/docs/devices/hub-connected/test-your-driver/

We also have tons of examples in the repo. In general, devices that are just fingerprint adds and do not change any code don't need new unit tests since they're not using any new code, but for devices where there _are_ code changes, we like to have tests of that logic so that we can avoid changing it with any future changes.

**Comment by sengled-desi** [comment]
Hey @greens  @lelandblue Since we don't have a test environment here, can you please help us test it? Our other products (pull: #724) are the same as the one I submitted, and there is no problem with those ones, so I think there should be no problem with this submission.

**Comment by greens** [comment]
@sengled-desi These two links should help:

https://developer.smartthings.com/docs/devices/hub-connected/set-up-dev-env
https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/releases

**Comment by sengled-desi** [comment]
@greens   It’s difficult for us to setup the Unit Test environment for E1M-G7H. so is there any other way we can move forward ?

**Comment by lelandblue** [comment]
@sengled-desi  Let me help clarify, we do not believe you need to set up an additional test environment. 

Instead, we are asking for additional code to be written in the form of unit test. The examples that are linked above and that were provided by Andrew are in a .text file. We do require that **unit tests** are written to cover this code as your pull request stands today since you are adding a sub-driver in the init file. 

We can help make sure that you have the information to move forward. Please let me know if you have questions.

**Comment by sengled-desi** [comment]
@greens @lelandblue    The .text file in this link you said, I didn’t find it in front. You can directly provide me with a similar example file.

**Comment by lelandblue** [comment]
Hello @sengled-desi Thank you for your communication. 

Here are some test file examples that are in the Zigbee Motion Sensor Driver that you can use as an example: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/tree/main/drivers/SmartThings/zigbee-motion-sensor/src/test

Here is also some test file examples that are for the Zigbee Contact Sensor. 
https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/tree/main/drivers/SmartThings/zigbee-contact/src/test

Each of the test files, coordinate with the init files of those devices. Please let me know if you have some questions still.

**Comment by sengled-desi** [comment]
@lelandblue @greens    Hello, let me ask two questions: 
1. It is our battery power as the product that we previously certified to use the battery percentage attribute. Whether we do n’t need to add Sub-Driver. Just help us add product support. 
2. 3.If The Sub-Driver is a Mandatory Requirement, then is my test driver uploaded directly on this page or submitted through github

**Comment by greens** [comment]
@sengled-desi our issues are with this:

```
/home/runner/work/SmartThingsEdgeDrivers/SmartThingsEdgeDrivers/drivers/SmartThings/zigbee-motion-sensor/src/sengled/init.lua | 86% | ❌
-- | -- | --
```
and this:
```
/home/runner/work/SmartThingsEdgeDrivers/SmartThingsEdgeDrivers/drivers/SmartThings/zigbee-contact/src/sengled/init.lua | 86% | ❌
-- | -- | --
```

**Comment by sengled-desi** [comment]
@lelandblue @greens    Hello, the test file has been added.  If you still can't test it, can you provide me with a sample test file? I have a lot of content not very clearly

File: drivers/SmartThings/zigbee-contact/src/test/test_sengled_contact_sensor.lua [hunk 3]
```diff
@@ -0,0 +1,98 @@
+-- Copyright 2022 SmartThings
```
Comment (greens) [inline drivers/SmartThings/zigbee-contact/src/test/test_sengled_contact_sensor.lua:1]: date

File: drivers/SmartThings/zigbee-motion-sensor/src/test/test_sengled_motion.lua [hunk 4]
```diff
@@ -0,0 +1,98 @@
+-- Copyright 2022 SmartThings
```
Comment (greens) [inline drivers/SmartThings/zigbee-motion-sensor/src/test/test_sengled_motion.lua:1]: date

**Comment by greens** [comment]
@sengled-desi looks good and now your coverage is at 100% so it's not shown in the digest. Thank you.

**Comment by lelandblue** [comment]
Hello @sengled-desi please update the copywrite dates on the new .test files and then we can consider this review complete. 

The PR will not be merged until the device completes testing.

**Comment by sengled-desi** [comment]
@lelandblue  Hello, The copyright date on the .test file has been updated.

**Review by greens — APPROVED** [review]

**Pull request merged**